src/dynamics_tendencies.jl

```diff
@@ -0,0 +1,216 @@
+"""
+Compute the spectral tendency of the surface pressure logarithm
+"""
+function surface_pressure_tendency!(Prog::PrognosticVariables{NF}, # Prognostic variables
+                                   Diag::PrognosticVariables{NF}, # Diagnostic variables
+                                   l2::Int,                       # leapfrog index 2 (time step used for tendencies)
+                                   C::Constants{NF}
+                                   ) where {NF<:AbstractFloat}
+
+
+    @unpack pres_surf= Prog 
+    @unpack pres_surf_tend = Diag.Tendencies
+
+    @unpack u_grid,v_grid,div_grid = Diag.gridvars
+    @unpack u_mean,v_mean,d_mean,dumc,px,py = Diag.miscvars
+    @unpack nlev, dhs = C 
+
+    #Initialise mean fields. 
+    u_mean[:,:] = 0.0
+    v_mean[:,:] = 0.0
+    d_mean[:,:] = 0.0 #d-mean is calculated here but actually used in vertical_velocity_tendency!()
+
+    #..and calculate values
+    for k in 1:nlev
+        u_mean += u_grid[:,:,k]*dhs[k] 
+        v_mean += v_grid[:,:,k]*dhs[k]
+        d_mean += div_grid[:,:,k]*dhs[k]
+    end
+
+    #Now use the mean fields
+    grad!(pres_surf[:,:,l2], dumc[:,:,2], dumc[:,:,3])
+    px = gridded(dumc[:,:,2]*3600, scale=true)
```

Comment:
This still uses `gridded` ...

Reply:
Marking resolved. Agreed in Slack to use `gridded`/ `spectral`